calico/felix/fetcd.py

```diff
@@ -267,12 +288,13 @@ def watch_etcd(self, update_splitter):
 
                 _log.debug("Response action: %s, key: %s",
                            response.action, response.key)
-                if response.action not in ("set", "create"):
-                    # FIXME: this check is over-broad.
+                if (response.action not in ("set", "create") and
+                        any((response.key.startswith(pfx) for pfx in
+                             PREFIXES_TO_RESYNC_ON_CHANGE))):
                     # It's purpose is to catch deletions of whole directories
                     # or other operations that we're not expecting.
-                    _log.warning("Unexpected action %s to %s; triggering "
-                                 "resync.", response.action, response.key)
+                    _log.warning("Unexpected event: %s; triggering resync.",
```

Comment:
No, this case actually catches delete for, say a whole workload or a whole host right now.  In those cases, we can't yet handle it gracefully so we have to resync.